lib/graphql/batch/query_group.rb

```diff
@@ -1,35 +1,48 @@
 module GraphQL::Batch
   class QueryGroup < QueryContainer
-    def initialize(queries, &block)
-      @pending_queries = queries.dup
-      @pending_queries.each do |query|
-        query.query_listener = self
+    def initialize(queries)
+      raise ArgumentError, "QueryGroup.new no longer takes a block, use #then instead" if block_given?
+      super(self)
+      @pending_queries = Set.new
+      @results = []
+
+      queries.each_with_index do |query, i|
+        if query.is_a?(QueryContainer)
+          @pending_queries.add(query)
+        else
+          @results[i] = query
```

Comment:
Does ruby just automatically expand the array when assigning to an index past the end?